src/hooks/useProducts.ts

```diff
@@ -15,25 +16,34 @@ export interface errors {
 }
 
 interface UseProductsResult {
-  products: UseQueryResult<ProductDto[]>;
+  products: ProductDto[];
+  productsLoading: boolean;
+  error: Error | null;
   createProduct: UseMutationResult<ProductDto, productProblemDetails, CreatedProductDto>;
   removeProduct: UseMutationResult<void, ProblemDetails, number>;
 }
 
-export const useProducts = (): UseProductsResult => {
-  const products = useQuery<ProductDto[]>(['items'], productService.getAll);
+export const useProducts = (filtered = false): UseProductsResult => {
+  const { data: products, isLoading, error, refetch } = useQuery<ProductDto[]>('products', productService.getAll);
+
+  const selectedProducts = useMemo(() => {
+    if (filtered) {
+      return products?.filter((product) => !product.retired || !product.category) ?? [];
+    }
+    return products ?? [];
+  }, [products, filtered]);
```

Comment:
filtering can be done in the `useQuery` hook, I think same could be done with `useCategories`
```
const products = useQuery<ProductDto[]>(['items'], productService.getAll, {
  select: (data) => (filtered ? data.filter((product) => !product.retired || !product.category) : data),
});
```

Reply:
Yeah that looks good, just added now. Thanks man